packages/dashboard/src/customization/widgets/queryWidget/multiQueryWidget.tsx

```diff
@@ -1,44 +1,58 @@
 import React, { ReactNode } from 'react';
 import { useDrop } from 'react-dnd';
 
-import { SiteWiseAssetQuery } from '@iot-app-kit/source-iotsitewise';
-
 import { useWidgetActions } from '../../hooks/useWidgetActions';
 import { ItemTypes } from '~/components/dragLayer/itemTypes';
 import { QueryWidget } from '../types';
 import { assignDefaultStyles } from '../utils/assignDefaultStyleSettings';
 import './queryWidget.css';
 import { mergeAssetQueries } from '~/util/mergeAssetQueries';
+import { ExtendedPanelAssetSummary } from '~/components/resourceExplorer/nextResourceExplorer';
 
 /**
  *
  * HOC widget component for handling drag and drop of a widget that can have multiple assets per query
  *
  */
-const MultiQueryWidgetComponent: React.FC<QueryWidget & { children: ReactNode }> = ({ children, ...widget }) => {
+const defaultOnDropAssetToWidget = (
+  assetSummary: ExtendedPanelAssetSummary,
+  widget: QueryWidget,
+  update: (widget: QueryWidget) => void
+) => {
+  const { queryAssetsParam = [] } = assetSummary;
+  const asset = queryAssetsParam[0];
+
+  const updatedWidget = {
+    ...widget,
+    properties: {
+      ...widget.properties,
+      queryConfig: {
+        ...widget.properties.queryConfig,
+        query: {
+          assets: mergeAssetQueries(widget.properties.queryConfig.query?.assets || [], asset),
+        },
+      },
+    },
+  };
+
+  update(assignDefaultStyles(updatedWidget));
+};
+const MultiQueryWidgetComponent: React.FC<
+  QueryWidget & {
+    children: ReactNode;
+    onDropAsset?: (
```

Comment:
I think this can just be a function with the assetSummary and Widget as a parameter and returns a Widget? we can call this in the MultiQueryWidgetComponent drop callback.

```drop: (summary: ExtendedPanelAssetSummary) => update(onDropAsset(summary, widget))```